File: hackerrank/warmup/cut-the-sticks/cut-the-sticks.py
#!/usr/bin/env python

raw_input();

input = raw_input().split();

sticks = [int(x) for x in input];

cuts = len(sticks);
while cuts > 0:
    print cuts
    smaller = min(sticks);
    sticks = [x - smaller for x in sticks if x > smaller];
    cuts = len(sticks);

File: hackerrank/warmup/cut-the-sticks/cut-the-sticks.rb
#!/usr/bin/env ruby

gets

sticks = gets.split.map {|x| x.to_i}

cuts = sticks.size

while (cuts > 0)
  puts cuts
  min = sticks.min
  sticks.delete_if {|a| a <= min}
  sticks = sticks.map {|a| a - min}
  cuts = sticks.size
end
